### PR TITLE
cors: set access-control-expose-headers on actual CORS request

### PR DESCRIPTION
**Pull request by stanley-cheung**
*Description*: Set the `access-control-expose-headers` response header on the actual CORS request (not preflight). Currently, if I set a custom header from the upstream server, I can't retrieve the value of that header from the browser getResponseHeaders() API because the `access-control-expose-headers` response header is only set on the preflight OPTIONS request, not the actual POST request.
*Risk Level*: low
*Testing*: bazel test /test/...
*Docs Changes*: None
*Release Notes*: None

Signed-off-by: Stanley Cheung <stanleycheung@google.com>

### CODE REVIEW

**Comment by stanley-cheung** [comment]
Maintainers: this is the first time I try to submit a PR. I noticed the 2 CI tests failures. How should I proceed? Any tips on how to reproduce that locally? Could that be flakes? My change seems unrelated. Please advise. Thanks!

**Comment by dio** [comment]
@stanley-cheung you can try to do master merge, hopefully, that will heal the `echo_integration_test` flakiness.

File: test/extensions/filters/http/cors/cors_filter_integration_test.cc [hunk 1]
```diff
@@ -66,6 +66,15 @@ class CorsFilterIntegrationTest : public HttpIntegrationTest,
...
             cors->add_allow_origin_regex(".*\\.envoyproxy\\.io");
           }
+
+          {
+            auto* route = virtual_host->add_routes();
+            route->mutable_match()->set_prefix("/cors-expose-headers");
+            route->mutable_route()->set_cluster("cluster_0");
+            auto* cors = route->mutable_route()->mutable_cors();
+            cors->add_allow_origin("test-origin-1");
+            cors->set_expose_headers("custom-header-1");
```
Comment (dio) [inline test/extensions/filters/http/cors/cors_filter_integration_test.cc]: Nit, can we have a comma-separated list of headers here (e.g. `X-My-Custom-Header, X-Another-Custom-Header`)? Just to make a point that we can express setting multiple headers by doing so.

**Comment by codesuki** [comment]
After re-reading the rfc I agree this is correct.

https://www.w3.org/TR/cors/#resource-requests

Also expose headers should be removed from the preflight request since it's not noted in 6.2.

Additionally I found some points where a preflight request should probably terminate early which it's not doing yet. I'll follow up on that later this week.

File: test/extensions/filters/http/cors/cors_filter_integration_test.cc (hunk 1 above)
Comment (stanley-cheung) [reply test/extensions/filters/http/cors/cors_filter_integration_test.cc]: @dio, thanks for the review. I have updated the test with multiple expose-headers.

Should I also take out the setting of `access-control-expose-headers` in the `OPTIONS` request in this same PR?

**Comment by stanley-cheung** [comment]
@codesuki Thanks for the review! I have updated the PR to not set `access-control-expose-headers` in the response to the `OPTIONS` request.

**Pull request merged**